pkg/registry/ggcr_image.go

```diff
@@ -0,0 +1,109 @@
+package registry
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pkg/errors"
+)
+
+type GoContainerRegistryImage struct {
```

Comment:
can we rename this file/struct ... maybe RemoteRegistryImage?

Reply:
My thought is the consumers of this object will consume it through the RemoteImage interface. What makes this struct unique is that it implements that interface using  GoContainerRegistry library.